src/query/formats/src/output_format/json.rs

```diff
@@ -105,6 +95,18 @@ impl OutputFormat for JSONOutputFormat {
             .iter()
             .map(|f| f.name().to_string())
             .collect::<Vec<String>>();
+
+        let mut res = if self.first_block {
+            self.first_block = false;
+            let mut buf = b"{".to_vec();
+            buf.extend_from_slice(self.format_schema()?.as_ref());
+            buf.extend_from_slice(format!(",\"rows\":{}", rows.len()).as_bytes());
```

Comment:
rows should be accumulated here and added in `finalize`

Reply:
fixed